### DISCUSSION TITLE
Pyarmor CI license not working in YAML based pipeline

### DISCUSSION THREAD

**Discussion by Dhruv-two** [dashingsoft/pyarmor#1993] [Q&A]
My pyarmor CI licence worked when i ran it on a docker image through dockerfile commands. However, it didn't work on another repo which was deploying through yaml pipelines on azure. How to resolve this issue? Should i add something in the yaml? What constitutes a "CI pipeline" for pyarmor ci license? like how does it detect if it is running in a normal VM or a pipeline? Even when I am running in YAML CI pipeline, it's not working.

**Comment by jondy** [comment]
Is there `/dev/disk` in yaml pipeline on azure?

**Reply by DAR6969** [reply]
it is there on the build agent but not sure in the pipeline

**Comment by jondy** [comment]
I alos have no idea about azure, but just add one shell command in the pipeline
```
ls /dev/disk
```

And check the output of this command

**Comment by jondy** [comment]
CI/License now only is used in no-disk system, it doesn't work if there is disk in running container.
